### ISSUE TITLE
`Asset::getUrl($transform, false)` sometimes uses incorrect domain in headless mode

### ISSUE THREAD

**Issue by FreekVR** [craftcms/cms#7387]
### Description
We have a custom controller within a Craft instance which runs in headless mode. 

Our frontend runs on a different domain from the CMS. As such, the url for the default site is set to something like 'http://localhost:3000' locally.

It seems that the getUrl() function for assets will sometimes use the default site URL rather than the domain under which the CMS runs, so we get a non-functional URL like https://localhost:3000/index.php?p=actions/assets/generate-transform&transformId=29 instead of https://cms-domain.test/index.php?p=actions/assets/generate-transform&transformId=29

I was able to work around this by setting the "immediate" param to `true` which DOES return the correct URL to the generated asset.

### Steps to reproduce

1. Add a custom controller somewhere in your headless craft app.
2. Return an asset with a code similar to this:
```
public function actionImage(int $assetId): Response
  $transform = 'thumbnail';
  $asset = Craft::$app->getAssets()->getAssetById($assetId);
  
  return $this->asJson([
    'src' => $asset->getUrl($transform, true),
    'width' => $asset->getWidth($transform),
    'height' => $asset->getHeight($transform),
  ]);
}
``` 

### Additional info

- Craft version: 3.6.0-RC3
- PHP version: 7.4.x
- Database driver & version: MySQL 8
- Plugins & versions: n/a

**Comment by brandonkelly** [comment]
What is your volume’s Base URL setting set to?

**Comment by kara-todd** [comment]
@FreekVR this may be the same issue I [was running into](https://github.com/craftcms/cms/issues/6597#issuecomment-759146497) and [resolved in the next 3.6 release](https://github.com/craftcms/cms/commit/52035a51e7fcfe2bc5e78cc97325d742b29b655a)

**Comment by FreekVR** [comment]
Hey @kara-todd yeah that's exactly what I was running into!

@brandonkelly the base URL was set to “@web/storage/" or something similar, but it was always using the default site url regardless of what I put here, I suppose the same issue as Kara mentioned there.

**Comment by brandonkelly** [comment]
Ah yeah makes sense. So should definitely be fixed for you in 3.6 🎉
